### ISSUE TITLE
Lots of variables becoming Unknown in v3.19.7 when the car goes to sleep

### ISSUE THREAD

**Issue by j4m35** [alandtse/tesla#843]
I just installed v3.19.7 and after the car goes to sleep, the following is an incomplete list of variables that will become unknown:
battery
asleep
location tracker
arrival time
time charge complete
energy added

When I revert back to v3.19.5, the variables will remain when the car goes to sleep.

**Comment by andreaconfa** [comment]
+1, same issue here

**Comment by Thlerry** [comment]
+1, same issue for me. How do you revert to the older working version ? Is there a way to downgrade ?

**Comment by andreagenovese** [comment]
same problem

**Comment by skipishere** [comment]
Could you supply some logs and include information such as Car model and software version?

For my Model 3, I've only noticed the Odometer recently going Unknown when it goes to sleep, otherwise my battery, location, energy added still report the last known value.

For `arrival time` and `time charge complete` these sound sensible to be Unknown as you've arrived/not driving and charging won't give you a value unless the car is charging (which then it would be awake)

**Comment by andreaconfa** [comment]
> Could you supply some logs and include information such as Car model and software version?
> 
> For my Model 3, I've only noticed the Odometer recently going Unknown when it goes to sleep, otherwise my battery, location, energy added still report the last known value.
> 
> For `arrival time` and `time charge complete` these sound sensible to be Unknown as you've arrived/not driving and charging won't give you a value unless the car is charging (which then it would be awake)

Hi.
how can i retrive the logs?

My car is Model 3 and i have two of them used in the integration. Both with the same issue

**Comment by andreaconfa** [comment]
I have a lot of errors regarding the tesl aintegration in my log:
`2024-01-29 14:21:34.178 ERROR (MainThread) [homeassistant.components.automation.tesla_andrea_refresh_integrazione_veicolo] Tesla Andrea Refresh Integrazione Veicolo: Error executing script. Unexpected error for call_service at pos 2:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 476, in _async_step
await getattr(self, handler)()
File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 713, in _async_call_service_step
response_data = await self._async_run_long_action(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 675, in _async_run_long_action
return long_task.result()
^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/core.py", line 2149, in async_call
response_data = await coro
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/core.py", line 2186, in _execute_service
return await target(service_call)
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 272, in handle_service
return await service.entity_service_call(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 882, in entity_service_call
single_response = await _handle_entity_call(
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 952, in _handle_entity_call
result = await task
^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
await self.async_press()
File "/config/custom_components/tesla_custom/button.py", line 84, in async_press
await self.update_controller(wake_if_asleep=True, force=True)
File "/config/custom_components/tesla_custom/base.py", line 82, in update_controller
await self.coordinator.controller.update(
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 734, in update
cars = await self.get_vehicles()
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 300, in get_vehicles
return (await self.api("VEHICLE_LIST"))["response"]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1319, in api
return await self.__post_with_retries_except_unavailable(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 88, in async_wrapped
return await fn(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 47, in __call__
do = self.iter(retry_state=retry_state)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 325, in iter
raise retry_exc.reraise()
^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 158, in reraise
raise self.last_attempt.result()
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 449, in result
return self.__get_result()
^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
raise self._exception
File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 50, in __call__
result = await fn(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1351, in __post_with_retries_except_unavailable
return await self.__connection.post(command, method=method, data=data, url=url)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 165, in post
return await self.__open(url, method=method, headers=self.head, data=data)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 217, in __open
raise TeslaException(resp.status_code)
teslajsonpy.exceptions.TeslaException`

**Comment by Thlerry** [comment]
Tesla Model 3 as well running 2020.44.30.8

For instance binary_sensor.asleep in HA is reporting status 'off' (incorrectly) when chargeHQ is reporting the car as asleep.
It seems that when you restart HA, it picks up the new value, but then doesn't manage to update it after, so the sensor remains with the old state when the car status actually changes.

I see the following errors in the logs:

Home Assistant Core
Unexpected error fetching tesla_custom data: [Errno -3] Try again
12:25:25 AM – (ERROR) Tesla Custom Integration (custom integration)

Error fetching tesla_custom data: Error communicating with API:
January 29, 2024 at 4:55:41 PM – (ERROR) Tesla Custom Integration (custom integration)

**Comment by andreaconfa** [comment]
Yes, i think that the "unknown" values is shown when HA is restarted.

**Comment by andreaconfa** [comment]
Also, if i click on "Update Data" button, no update is made. 

LOG:

> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 238, in handle_call_service
>     response = await hass.services.async_call(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/core.py", line 2149, in async_call
>     response_data = await coro
>                     ^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/core.py", line 2186, in _execute_service
>     return await target(service_call)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 272, in handle_service
>     return await service.entity_service_call(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 882, in entity_service_call
>     single_response = await _handle_entity_call(
>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 952, in _handle_entity_call
>     result = await task
>              ^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
>     await self.async_press()
>   File "/config/custom_components/tesla_custom/button.py", line 84, in async_press
>     await self.update_controller(wake_if_asleep=True, force=True)
>   File "/config/custom_components/tesla_custom/base.py", line 82, in update_controller
>     await self.coordinator.controller.update(
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 734, in update
>     cars = await self.get_vehicles()
>            ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 300, in get_vehicles
>     return (await self.api("VEHICLE_LIST"))["response"]
>             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1319, in api
>     return await self.__post_with_retries_except_unavailable(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 88, in async_wrapped
>     return await fn(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 47, in __call__
>     do = self.iter(retry_state=retry_state)
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 314, in iter
>     return fut.result()
>            ^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 449, in result
>     return self.__get_result()
>            ^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
>     raise self._exception
>   File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 50, in __call__
>     result = await fn(*args, **kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1351, in __post_with_retries_except_unavailable
>     return await self.__connection.post(command, method=method, data=data, url=url)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 165, in post
>     return await self.__open(url, method=method, headers=self.head, data=data)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 217, in __open
>     raise TeslaException(resp.status_code)
> teslajsonpy.exceptions.TeslaException

**Comment by andreagenovese** [comment]
Mee too, update not work.

**Comment by j4m35** [comment]
> Could you supply some logs and include information such as Car model and software version?
> 
> For my Model 3, I've only noticed the Odometer recently going Unknown when it goes to sleep, otherwise my battery, location, energy added still report the last known value.
> 
> For `arrival time` and `time charge complete` these sound sensible to be Unknown as you've arrived/not driving and charging won't give you a value unless the car is charging (which then it would be awake)

Hi,

I have a Model 3 running 2023.44.30.9.

I understand what you mean by 'time charge complete' sounds sensible, but usually 'arrival time' stays with the last updated value for me. Anyway, I have traced the time when the values became Unknown to this log entry:
`
2024-01-28 20:25:58.853 ERROR (MainThread) [custom_components.tesla_custom] Unexpected error fetching tesla_custom data: 'drive_state'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/tesla_custom/__init__.py", line 442, in _async_update_data
    data = await controller.update(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 817, in update
    return any(await asyncio.gather(*tasks))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 640, in _get_and_process_car_data
    != response["drive_state"]["shift_state"]
       ~~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'drive_state'
`

There are many instances of this error entry each time it tries to refresh the data (on v3.19.7) whilst the car is sleeping.

**Comment by j4m35** [comment]
Just to add, this is what it looks like in the UI log when the car goes to sleep:
```
IIU Polling became unavailable
9:44:41 AM - 13 hours ago
IIU Sentry mode became unavailable
9:44:41 AM - 13 hours ago
IIU Heated steering became unavailable
9:44:41 AM - 13 hours ago
IIU Destination location tracker was detected at unavailable
9:44:41 AM - 13 hours ago
IIU Location tracker was detected at unavailable
9:44:41 AM - 13 hours ago
IIU User present became unavailable
9:44:41 AM - 13 hours ago
IIU Scheduled departure became unavailable
9:44:41 AM - 13 hours ago
IIU Scheduled charging became unavailable
9:44:41 AM - 13 hours ago
IIU Windows became unavailable
9:44:41 AM - 13 hours ago
IIU Doors became unavailable
9:44:41 AM - 13 hours ago
IIU Charging became unavailable
9:44:41 AM - 13 hours ago
IIU Charger became unavailable
9:44:41 AM - 13 hours ago
IIU Asleep became unavailable
9:44:41 AM - 13 hours ago
IIU Online became unavailable
9:44:41 AM - 13 hours ago
IIU Parking brake became unavailable
9:44:41 AM - 13 hours ago
IIU Hvac (climate) system became unavailable
9:44:41 AM - 13 hours ago
IIU Data last update time became unavailable
9:44:41 AM - 13 hours ago
IIU Arrival time became unavailable
9:44:41 AM - 13 hours ago
IIU Time charge complete became unavailable
9:44:41 AM - 13 hours ago
IIU Shift state became unavailable
9:44:41 AM - 13 hours ago
IIU Doors was unlocked
9:33:16 AM - 13 hours ago
```

The car is parked at 9:33am, the car goes to sleep, and when v3.19.7 polls at 9:44am those are all the entries that turn unavailable.

**Comment by skipishere** [comment]
> Mee too, update not work.

For clicking the update see https://github.com/alandtse/tesla/issues/774

**Comment by jmcollin78** [comment]
Same problem for me this morning. All data are suddenly not available. Awaking the car with the wake up button make it available once again.

**Comment by LightBringer81** [comment]
Same here, if I reload the integration it works again with current data.

**Comment by garethcook** [comment]
Same here. Down-levelling to 3.9.15 for the moment

**Comment by Thlerry** [comment]
> Tesla Model 3 as well running 2023.44.30.8
> 
> For instance binary_sensor.asleep in HA is reporting status 'off' (incorrectly) when chargeHQ is reporting the car as asleep. It seems that when you restart HA, it picks up the new value, but then doesn't manage to update it after, so the sensor remains with the old state when the car status actually changes.
> 
> I see the following errors in the logs:
> 
> Home Assistant Core Unexpected error fetching tesla_custom data: [Errno -3] Try again 12:25:25 AM – (ERROR) Tesla Custom Integration (custom integration)
> 
> Error fetching tesla_custom data: Error communicating with API: January 29, 2024 at 4:55:41 PM – (ERROR) Tesla Custom Integration (custom integration)

Update on my side, I downgraded to 3.19.5 and it seems I still have the issue with the  binary_sensor.asleep not updating.
Could someone check if this sensor is working for them with 3.19.5 ?

**Comment by darek-margas** [comment]
> Update on my side, I downgraded to 3.19.5 and it seems I still have the issue with the binary_sensor.asleep not updating.
> Could someone check if this sensor is working for them with 3.19.5 ?

Mine works correctly:
![image](https://github.com/alandtse/tesla/assets/71180450/150bf282-6c5c-4d71-adcd-25256eb9a071)
3.19.7

**Comment by EliteAppleHacks** [comment]
> Mine works correctly:
> ![image](https://private-user-images.githubusercontent.com/71180450/301394731-150bf282-6c5c-4d71-adcd-25256eb9a071.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NjU0MDcsIm5iZiI6MTcwNjc2NTEwNywicGF0aCI6Ii83MTE4MDQ1MC8zMDEzOTQ3MzEtMTUwYmYyODItNmM1Yy00ZDcxLWFkY2QtMjUyNTZlYjlhMDcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAxVDA1MjUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzYTFiZjQxYjg2NWZmZDU5MGU2YzMxNzgyYzA1MTY3NGZiYmFkMTE1NDBiN2M3M2FjOGFkZGFhMDNjMWVlZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Xmnl0HaKFZEuzvNHhH2uOTSaQYC13gV-268-BNXkDyM)
> 3.19.7

both 3.19.5 and .7 not working for me. My car was just asleep and i woke it up with my HA button and still not reporting. Car version 2023.44.30.8

![image](https://github.com/alandtse/tesla/assets/13251160/a78cde43-3f0d-4e84-b9f6-fb60740d68fa)

**Comment by Thlerry** [comment]
> 2023.44.30.8

Which version of Tesla OS are you running in the car ? Are you on 2023.44.30.8 as well ?

**Comment by Thlerry** [comment]
> > Update on my side, I downgraded to 3.19.5 and it seems I still have the issue with the binary_sensor.asleep not updating.
> > Could someone check if this sensor is working for them with 3.19.5 ?
> 
> Mine works correctly: ![image](https://private-user-images.githubusercontent.com/71180450/301394731-150bf282-6c5c-4d71-adcd-25256eb9a071.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NjY1MDIsIm5iZiI6MTcwNjc2NjIwMiwicGF0aCI6Ii83MTE4MDQ1MC8zMDEzOTQ3MzEtMTUwYmYyODItNmM1Yy00ZDcxLWFkY2QtMjUyNTZlYjlhMDcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAxVDA1NDMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiOWNjOWZjYzcxYjljOTUxNDlmZWM5ZmJhOTVlMzA2NTQ3YWM4MWM2YzhmMzA4OWJhYzM4ZmJiMjE1YjQ2OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UT-voyw37y4iElUkPNGpV3jK7Br7_RCHO9orwefQ9vk) 3.19.7

Which version of Tesla OS are you running in the car ? Are you on 2023.44.30.8 as well ?

**Comment by Thlerry** [comment]
> > Update on my side, I downgraded to 3.19.5 and it seems I still have the issue with the binary_sensor.asleep not updating.
> > Could someone check if this sensor is working for them with 3.19.5 ?
> 
> Mine works correctly: ![image](https://private-user-images.githubusercontent.com/71180450/301394731-150bf282-6c5c-4d71-adcd-25256eb9a071.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NjY1MDIsIm5iZiI6MTcwNjc2NjIwMiwicGF0aCI6Ii83MTE4MDQ1MC8zMDEzOTQ3MzEtMTUwYmYyODItNmM1Yy00ZDcxLWFkY2QtMjUyNTZlYjlhMDcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAxVDA1NDMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiOWNjOWZjYzcxYjljOTUxNDlmZWM5ZmJhOTVlMzA2NTQ3YWM4MWM2YzhmMzA4OWJhYzM4ZmJiMjE1YjQ2OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UT-voyw37y4iElUkPNGpV3jK7Br7_RCHO9orwefQ9vk) 3.19.7

Which version of Tesla OS are you running in the car ? Are you on 2023.44.30.8 as well ?

**Comment by andreagenovese** [comment]
Exception If I try to manual update data from integration when car in sleep.

Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:238
Integration: Home Assistant WebSocket API ([documentation](https://www.home-assistant.io/integrations/websocket_api), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+websocket_api%22))
First occurred: 08:26:45 (1 occurrences)
Last logged: 08:26:45

[139755856777920]
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 238, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2149, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2186, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 272, in handle_service
    return await service.entity_service_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 882, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 952, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/button/__init__.py", line 132, in _async_press_action
    await self.async_press()
  File "/config/custom_components/tesla_custom/button.py", line 84, in async_press
    await self.update_controller(wake_if_asleep=True, force=True)
  File "/config/custom_components/tesla_custom/base.py", line 82, in update_controller
    await self.coordinator.controller.update(
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 734, in update
    cars = await self.get_vehicles()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 300, in get_vehicles
    return (await self.api("VEHICLE_LIST"))["response"]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1319, in api
    return await self.__post_with_retries_except_unavailable(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 88, in async_wrapped
    return await fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 47, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/tenacity/__init__.py", line 314, in iter
    return fut.result()
           ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.11/site-packages/tenacity/_asyncio.py", line 50, in __call__
    result = await fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/controller.py", line 1351, in __post_with_retries_except_unavailable
    return await self.__connection.post(command, method=method, data=data, url=url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 165, in post
    return await self.__open(url, method=method, headers=self.head, data=data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/teslajsonpy/connection.py", line 217, in __open
    raise TeslaException(resp.status_code)
teslajsonpy.exceptions.TeslaException

**Comment by darek-margas** [comment]
> Which version of Tesla OS are you running in the car ? Are you on 2023.44.30.8 as well ?

2023.44.30.9

**Comment by andreagenovese** [comment]
> > Which version of Tesla OS are you running in the car ? Are you on 2023.44.30.8 as well ?
> 
> 2023.44.30.9

My version is 2023.44.30.9

**Comment by j4m35** [comment]
Guys, I think there are 2 issues getting muddled together in this ticket.

1. Asleep no longer changes to ON even on v3.19.5 even when it used to. And this is most likely because of Tesla's API change: https://github.com/zabuldon/teslajsonpy/issues/448

2. The main issue I am talking about is this: https://github.com/alandtse/tesla/issues/843#issuecomment-1914827590 where on v3.19.7 a lot of values will turn to Unknown instead of staying as their previous values in v3.19.5.

**Comment by goxofy** [comment]
same as

**Comment by chezpaul** [comment]
I'm lazy, I don't have my logs but same for me. "Unknowns" all over the place with 3.19.7

**Comment by louliangsheng** [comment]
same as

**Comment by Nessy1970** [comment]
At first glance, fixed for me with version 3.19.8

**Comment by jmcollin78** [comment]
3.19.8 fixed the odometer to be undefined. 🎉

**Comment by skipishere** [comment]
Making this closed as version 3.19.8 appears to fix this. If people still have issues feel free to reopen the issue/comment
